DataDefinitions/Ship.cs

```diff
@@ -357,6 +375,44 @@ public string CoriolisUri()
             return null;
         }
 
+        public string EDShipyardUri()
+        {
+            if (raw != null)
+            {
+                // Generate an EDShipyard import URI to retain as much information as possible
+                
+                string uri = "http://www.edshipyard.com/";
+
+                // Take the ship's JSON, gzip it, then turn it in to base64 and attach it to the base uri
+                string unescapedraw = raw.Replace(@"\""", @"""");
+                var bytes = Encoding.UTF8.GetBytes(unescapedraw);
+                using (var streamIn = new MemoryStream(bytes))
+                using (var streamOut = new MemoryStream())
+                {
+                    using (var gzipStream = new GZipStream(streamOut, CompressionLevel.Optimal, true))
+                    {
+                        streamIn.CopyTo(gzipStream);
+                    }
+                    uri += "#/I=" + Uri.EscapeDataString(Convert.ToBase64String(streamOut.ToArray()));
+                }
+
+                // Add the ship's name
+                string bn;
```

Comment:
I know this got superseded, but as a tip you C# 6 lets you write this more elegantly as:
`string bn = name ?? $"{role} {model}";`